include/bitsery/adapter/buffer.h

```diff
@@ -261,6 +261,15 @@ class OutputBufferAdapter
     writeInternalImpl(data, size);
   }
 
+  TValue* writeInternalBuffer(size_t size)
+  {
+    const size_t newOffset = _currOffset + size;
+    maybeResize(newOffset, TResizable{});
+    _currOffset = newOffset;
+    return _beginIt + static_cast<diff_t>(_currOffset);
```

Comment:
Either I miss something, or this is a bug.
Instead of
```
    _currOffset = newOffset;
    return _beginIt + static_cast<diff_t>(_currOffset);
```
You should probably do like this:
```
    auto res = _beginIt + static_cast<diff_t>(_currOffset);
    _currOffset = newOffset;
    return res;
```